keps/XX-rbg-warmup/README.md

```diff
@@ -0,0 +1,209 @@
+
+## Summary
+
+This KEP proposes adding image and volume warmup capabilities to RoleBasedGroup (RBG). Cold start is a common problem for serving large language models(LLMs).
+Cold start latency can be reduced by pre-warming the model files and the container image before a serving instance starts. In RBG's design, upgradation can be done in a fine-grained manner (e.g. [Role Coordination](../30-role-coordination/README.md)). Image and volume warmup can be an additional step to speed up the upgrade process, avoiding resource utilization downgradation.
+
+## Motivation
+
+In a production environment, availability and GPU resource utilization are key metrics for large language model inference services. In-place update reduces time consumption by avoiding recreating Pods, but the cold start latency of the inference service (e.g. SGLang Worker) can still be high and thus affecting service availability. There are two key factors that affect the cold start latency:
+- The size of container's image size (e.g. SGLang v0.5.5 image size is about 27.5GB after decompression)
+- The size of large language model files (e.g. Total size of Qwen3-32B is about 61GB)
+
+Image and volume warmup can help reduce the cold start latency. By introducing an extra "warmup stage" before the upgrade process, the upgradation process can be more efficient. It takes less time for RoleBasedGroup to rolling update each inference engine instance, so both the service availability and GPU utilization will be improved. 
+
+Without WarmUp:
+![alt text](without-warmup.png)
+
+With WarmUp
+![alt text](with-warmup-stage.png)
+
+### Goals
+
+- Support WarmUp job that can be submitted by a user. For each node in the cluster, the job warms up images and volumes according to the job's configuration.
+
+### Non-Goals
+
+- Althgouh it's possible to upgrade RBG and warm up images & volumes in a coordinated way, co-design between WarmUp and RBG's upgradation process is not a goal in this enhancement. 
+
+## Proposal
+
+### Architecture
+
+![alt text](arch.png)
+
+The WarmUp Controller collects node information of all Pods under the target RBG. It deploys one WarmUp Pod to each node. Each WarmUp Pod contains:
+- N containers for image warmup, configured with image=<new_image> and a startup command set to "sleep infinity"
+- 1 container for volume warmup, which shares volumes with the inference service Pods of the RBG. The model files in the volume will be read sequentially after the container starts.
+
+When the Pod becomes ready, the node is considered warmed up successfully. If all nodes have completed the process, the WarmUp succeeds.
+
+### API Changes
+
+```yaml
+apiVersion: workloads.x-k8s.io/v1alpha1
+kind: RoleBasedGroupWarmup
+metadata:
+  name: warmup-demo
+  namespace: default
+spec:
+  policies:
+    parallelism: 10 # how many nodes will be warmed up at the same time
+    backoffLimit: 3 #
+    timeoutSeconds: 300
+    ttlSecondsAfterFinished: 86400
+  images:
```

Comment:
What is the relationship between images, volumes, and targets? Can the overall configuration be simplified? Is retaining only the target rbg configuration insufficient for preheating?